two_population_analysis/Snakefile

```diff
@@ -1,83 +1,398 @@
+        os.chdir(cur)
 
-rule fit_dadi_model_to_msprime_sfs:
+rule smcpp_plot:
     input:
-        sfs_files
+        rules.run_smcpp.output
     output:
-        "results_fits/homo_sapiens_OutOfAfrica/{demo_models}/model_fit_{fit_seeds}.pdf",
-        "results_fits/homo_sapiens_OutOfAfrica/{demo_models}/model_params_{fit_seeds}.txt"
-    threads: 4
-    run: dadi_utils.fit_dadi_model(sfs_files,output[0],output[1],wildcards.demo_models, wildcards.fit_seeds)
+        output_dir + "/{seeds}/trees.smc.gz.final.json.csv", output_dir + "/{seeds}/split/model.final.json", output_dir + "/{seeds}/split/joint.csv"
+    run:
+        # TODO get the generation time from std source
+        print(input)
+        smc.run_smcpp_plot(input[0], input[1], output[0], generation_time=generation_time)
+
+
+def ne_files_smcpp(wildcards):
+    return expand(output_dir + "/{seeds}/split/model.final.json",
+            seeds=seed_array, chrms=chrm_list)
+
+
+rule compound_smcpp:
+    input: ne_files_smcpp
+    output: output_dir + "/smcpp_estimated_Ne.png"
+    run: plots.plot_compound_smcpp( input, output[0])
+
+
+
+# ###############################################################################
+# Compile results and plot
+# ###############################################################################
+
+
+rule get_dadi_fsc_output:
+    input: expand(output_dir +  "/{seeds}/dadi_analysis/{demo_model}/model_params_{fit_seeds}.txt", fit_seeds=dadi_seed_array, seeds=seed_array, demo_model=dadi_model),
+           expand(output_dir +  "/{seeds}/fsc_analysis/run10/{demo_model}/{demo_model}.bestlhoods",seeds=seed_array,demo_model=fsc_model)
+
+    output: output_dir+"/{seeds}/dadi_analysis/dadi_results_sorted.txt",output_dir+"/{seeds}/fsc_analysis/fsc_results_sorted.txt"
+
+    run:
+        for seed in seed_array:
+            dadi_utils.get_dadi_output_IM(output_dir+"/"+str(seed)+ "/dadi_analysis/",dadi_model[0],dadi_seed_array,"dadi_results_sorted.txt")
+            fastsimcoal.get_fsc_output(output_dir+"/"+str(seed)+"/fsc_analysis/",10,"fsc_results_sorted.txt",fsc_model[0])
+
+rule get_best_dadi_fsc_runs:
+    input: expand(output_dir+"/{seeds}/dadi_analysis/dadi_results_sorted.txt", seeds=seed_array),
+           expand(output_dir+"/{seeds}/fsc_analysis/fsc_results_sorted.txt", seeds=seed_array),
+           expand(output_dir+"/{seeds}/split/joint.csv", seeds=seed_array)
+
+    output: output_dir + "/best_fsc_runs.txt", output_dir + "/best_dadi_runs.txt", output_dir + "/smcpp_runs.txt"
+
+    run:
+        dadi_utils.get_best_dadi_runs(output_dir, seed_array, output[1])
+        fastsimcoal.get_best_fsc_runs(output_dir, seed_array, output[0])
+        smc.get_smcpp_runs(output_dir, seed_array, output[2])
+
+
+rule plot_estimates:
+    input: rules.get_best_dadi_fsc_runs.output
+
+    output: output_dir + "/estimates_N_tdiv_dadi_fsc_smcpp.png", output_dir + "/estimates_mig_dadi_fsc.png"
+
+    run:
+        print("chrom list:" + chrm_list)
```

Comment:
this throws an error as `chrm_list` is a list and not a string. you can just remove this